core/scripts/IntegrationTests.js

```diff
@@ -0,0 +1,567 @@
+    for (let i = 0; i < numIterations; i++) {
+      process.stdout.write(i.toString() + ", ");
+      // pick the sponsor and token holder from their arrays
+      sponsor = sponsors[i % sponsors.length];
+      tokenHolder = tokenHolders[i % tokenHolders.length];
+
+      // STEP 1: creating position
+      const tokensOutstanding = await expiringMultiParty.totalTokensOutstanding();
+      const rawCollateral = await expiringMultiParty.rawTotalPositionCollateral();
+      const GCR = rawCollateral.mul(toBN(toWei("1"))).div(tokensOutstanding);
+      const collateralNeeded = baseNumTokens.mul(GCR).div(toBN(toWei("1")));
+
+      await expiringMultiParty.create(
+        { rawValue: collateralNeeded.toString() },
+        { rawValue: baseNumTokens.toString() },
+        { from: sponsor }
+      );
+      positionsCreated++;
+
+      // STEP 2: transferring tokens to the token holder
+      if (i % 2 == 1) {
+        await syntheticToken.transfer(tokenHolder, baseNumTokens.toString(), {
+          from: sponsor
+        });
+        tokenTransfers++;
+      }
+
+      // STEP 3: advancing time
+      const currentTime = await expiringMultiParty.getCurrentTime();
+      await expiringMultiParty.setCurrentTime(currentTime.add(timeOffsetBetweenTests));
+      await mockOracle.setCurrentTime(currentTime.add(timeOffsetBetweenTests));
+
+      // STEP 4.a: chance to liquidate position. 1 in 3 will get liquidated
+      if (i % 3 == 1 && runLiquidations) {
+        const positionTokensOutstanding = (await expiringMultiParty.positions(sponsor)).tokensOutstanding;
+        await expiringMultiParty.createLiquidation(
+          sponsor,
+          { rawValue: GCR.toString() },
+          { rawValue: positionTokensOutstanding.toString() },
+          { from: liquidator }
+        );
+
+        // get the liquidation info from the event. Used later on to withdraw by accounts.
+        const liquidationEvents = await expiringMultiParty.getPastEvents("LiquidationCreated");
+        const liquidationEvent = liquidationEvents[liquidationEvents.length - 1].args;
+
+        liquidationsObject.push({
+          sponsor: liquidationEvent.sponsor,
+          id: liquidationEvent.liquidationId.toString(),
+          disputed: false
+        });
+
+        // STEP 4.b) Chance to dispute the liquidation. 1 in 2 liquidations will get disputed
+        if (i % 2 == 1 && runDisputes) {
+          // Create the dispute request for the liquidation
+          await expiringMultiParty.dispute(liquidationEvent.liquidationId.toString(), liquidationEvent.sponsor, {
+            from: disputer
+          });
+
+          // Push a price into the oracle. This will enable resolution later on when the disputer
+          // calls `withdrawLiquidation` to extract their winnings.
+          const liquidationTime = await expiringMultiParty.getCurrentTime();
+          await mockOracle.pushPrice(constructorParams.priceFeedIdentifier, liquidationTime, disputePrice);
```

Comment:
Can you explain how it's determined whether a dispute is successful or not?

Reply:
in this implementation are disputes are successful based on the resolved DVM price.

Reply:
Wait, are you saying they are always successful in this implementation?

Reply:
The way the input numbers are defined, the dispute price results in in an invalid liquidation. As a result, the dispute is always successful when running in the tests.

Reply:
Gotcha.